src/marqo/tensor_search/add_docs.py

```diff
@@ -129,25 +129,40 @@ def threaded_download_and_preprocess_content(allocated_docs: List[dict],
 
                     elif (modality in [Modality.VIDEO, Modality.AUDIO] and is_no_index) or (
                             modality in [Modality.VIDEO, Modality.AUDIO] and is_unstructured_index) or (
-                            is_structured_index and media_field_types_mapping[field] in [FieldType.AudioPointer, FieldType.VideoPointer]):
-                            
+                            is_structured_index and media_field_types_mapping[field] in [FieldType.AudioPointer, FieldType.VideoPointer] and modality not in [Modality.TEXT, Modality.IMAGE]):
+
                         if marqo_index.model.properties.get('type') not in [
                             ModelType.LanguageBind] and modality not in marqo_index.model.properties.get(
                             'supported_modalities'):
                             media_repo[doc[field]] = UnsupportedModalityError(
                                 f"Model {marqo_index.model.name} does not support {modality}")
                             continue
+
+                        if is_structured_index:
+                            if modality is Modality.VIDEO and media_field_types_mapping[
+                            field] is not FieldType.VideoPointer:
+                                media_repo[doc[field]] = S2InferenceError(
+                                    f"Invalid audio file. Error processing media file {doc}, detected as video, but field type is not VideoPointer")
+                                continue
+
+                            if modality is Modality.AUDIO and media_field_types_mapping[
+                            field] is not FieldType.AudioPointer:
+                                media_repo[doc[field]] = S2InferenceError(
```

Comment:
Create a new error for this. S2InferenceError is too generic. Maybe `MediaMismatchError`?  Make sure the error is caught properly so it does not break the whole batch but just returns 400 for the document.

Reply:
This might affect a bunch of tests/logic we have, esp in tensor_search. This will require a bit more investigation. Not sure if it fits here. I think we can leave it at `S2InferenceError` for now

Reply:
done